content/en/observability_pipelines/processors/remap_ocsf/custom_configuration.md

```diff
@@ -0,0 +1,369 @@
+
+### Mapping lookup tables
+
+A lookup table names a series of values to compare against the source column, along with the
+resulting `value` that should be used if the comparison succeeds. The comparison value is specified as one of the following operations:
+
+| Name            | Description                                                                                     |
+|-----------------|-------------------------------------------------------------------------------------------------|
+| `contains: S`   | Matches if `value` is a string and contains `S`, which must also be a string.                   |
+| `equals: V`     | Matches if `value` is exactly the same as `V` including type.                                   |
+| `equals_source: P` | Matches if `value` is exactly the same as the field at `P`.                                  |
+| `matches: R`    | Matches if `value` is a string that matches the regular expression `R`.                        |
+| `not_matches: R`| Matches if `matches` above does not.                                                            |
+
+The table may also include a `default` value to be used if none of the comparisons succeed. If a `default` is not configured and none of the values match, the source value is passed through as is.
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+mapping:
+- dest: src_endpoint.os.type
+  source: client.userAgent.os
+  lookup:
+    table:
+    - contains: Windows
+      value: Windows
+    - equals: Linux
+      value: Linux
+    - matches: "^[Mm]ac *OS"
+      value: macOS
+    default: Unknown
+- dest: activity_name
+  source: createdDateTime
+  lookup:
+    table:
+    - equals_source: lastUpdatedDateTime
+      value: Create
+    default: Update
+```
+
+{{% /tab %}}
+{{% tab "JSON example" %}}
+
+```json
+{
+  "mapping": [
+    {
+      "dest": "src_endpoint.os.type",
+      "source": "client.userAgent.os",
+      "lookup": {
+        "table": [
+          {
+            "contains": "Windows",
+            "value": "Windows"
+          },
+          {
+            "equals": "Linux",
+            "value": "Linux"
+          },
+          {
+            "matches": "^[Mm]ac *OS",
+            "value": "macOS"
+          }
+        ],
+        "default": "Unknown"
+      }
+    },
+    {
+      "dest": "activity_name",
+      "source": "createdDateTime",
+      "lookup": {
+        "table": [
+          {
+            "equals_source": "lastUpdatedDateTime",
+            "value": "Create"
+          }
+        ],
+        "default": "Update"
+      }
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Mapping Functions
+
+A function applies an operation to the value extracted from the source, before assigning to the destination.
```

Comment:
Before assigning the extracted value to the destination? Is that correct?

Reply:
Correct.